client/state/state_manager.js

```diff
@@ -13,86 +13,108 @@ See the License for the specific language governing permissions and
 limitations under the License.
 ==============================================================================*/
 
-import * as sharedState from '/lib/shared_state.js';
-import Debug from '/lib/lame_es6/debug.js';
-const debug = Debug('wall:state_manager');
+import * as time from '../util/time.js';
+import {decodeInterpolator, SharedState} from '/lib/shared_state.js';
+import assert from '/lib/assert.js';
 
-class ClientSharedState extends sharedState.SharedState {
-  constructor(name, interpolator, network) {
-    super(name, interpolator);
-    this.network_ = network;
-    this.owner_ = network.id;
-  }
-  set(value, time) {
-    if (this.owner_ !== undefined && this.network_.id !== this.owner_) {
-      throw new Error('Attempted to set state for state owned by ' + this.owner_);
-    }
-    super.set(value, time);
-    this.network_.emit('newclientstateset',
-        { name: this.name_, value: value, time: time });
+
+class StateRecord {
+  constructor() {
+    this.state = {};
+    this.priorData = {};
+    this.clientClosedTime = Infinity;
+    this.serverClosedTime = Infinity;
+    this.lastUpdatedTime = time.now();
   }
 }
 
-// Describes something that tracks all sharedstate and provides methods for
-// the communication of that state across the network.
-export class StateManager {
-  constructor(network) {
-    // A map of tracked state name -> state variable.
-    this.trackedState_ = {};
-    this.network_ = network;
+function isClosedOrStale(state) {
```

Comment:
lol, should be a method on StateRecord, no?

Reply:
Oops, yeah.